src/perfrecord.cpp

```diff
@@ -190,3 +224,40 @@ void PerfRecord::sendInput(const QByteArray& input)
     Q_ASSERT(m_perfRecordProcess);
     m_perfRecordProcess->write(input);
 }
+
+bool PerfRecord::checkFilePermissions(const QString &filePath)
+{
+    QFileInfo outputFile(filePath);
+    QString username = currentUsername();
+
+    if (outputFile.owner() != username) {
+        QString sudoExe = sudoUtil();
+        if (sudoExe.isEmpty() || username.isEmpty()) {
+            return false;
+        }
+
+        QString chownCommand = sudoExe + QStringLiteral(" -u root chown ") +
+                               username + QLatin1Char(':') +
+                               username + QLatin1Char(' ') +
+                               filePath;
+        m_chownProcess->start(chownCommand);
+        m_chownProcess->waitForFinished();
+    }
+    return true;
```

Comment:
this should still return `outputFile.isReadable()`, no? I.e. potentially the `chown` command above fails?